ethereumj-core/src/main/java/org/ethereum/sharding/config/BeaconConfig.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.sharding.config;
+
+import org.ethereum.config.CommonConfig;
+import org.ethereum.config.SystemProperties;
+import org.ethereum.datasource.BatchSourceWriter;
+import org.ethereum.datasource.DbSource;
+import org.ethereum.datasource.WriteCache;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.manager.WorldManager;
+import org.ethereum.sharding.manager.ShardingWorldManager;
+import org.ethereum.sharding.service.ValidatorService;
+import org.ethereum.sharding.crypto.DepositAuthority;
+import org.ethereum.sharding.contract.DepositContract;
+import org.ethereum.sharding.crypto.UnsecuredDepositAuthority;
+import org.ethereum.sharding.util.Randao;
+import org.ethereum.sharding.service.ValidatorRepository;
+import org.ethereum.sharding.service.ValidatorServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * @author Mikhail Kalinin
+ * @since 21.07.2018
+ */
+@Configuration
+@Conditional(BeaconConfig.Enabled.class)
+public class BeaconConfig {
+
+    @Autowired
```

Comment:
It's very dangerous to use `@Autowired` in configuration, but if it works for you and it's prototype we could keep it

Reply:
Good note. I am not fond of Spring DI due to it's unpredictable behaviour in some cases.
Reading from `@Import` description:
> `@Bean` definitions declared in imported `@Configuration` classes should be accessed by using `@Autowired` injection. Either the bean itself can be autowired, or the configuration class instance declaring the bean can be autowired. The latter approach allows for explicit, IDE-friendly navigation between `@Configuration` class methods.

This standing doesn't literally say that beans autowired in configuration classes are always set before their first usage. But it could be concluded that way. Let's keep it this way unless something is broken.